staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/ratcheting.go

```diff
@@ -145,7 +145,7 @@ func (r *ratchetingValueValidator) getValidateOption() validate.Option {
 //
 // This call has a side-effect of populating it's `children` variable with
 // the explored nodes of the object tree.
-func (r *ratchetingValueValidator) Validate() *validate.Result {
+func (r *ratchetingValueValidator) Validate(new interface{}) *validate.Result {
```

Comment:
We don't care about new because we already know what it is as a consequence of doing correlation. It should be the same.

The reason it is required as a parameter is to conform to the `ValueValidator` interface so that it can be returned directly by `SubPropertyValidator` and `SubIndexValidator`. We also ignored `new` in the previous edition of the code so it is not a semantic change.

Reply:
I'm not fan of this kind of things, any chance we can make it so that we don't have to use "new" when we create the validator but use the value given as a parameter instead?

Reply:
It's not optimal, but I dont see a good way around it. I dont think would make sense for `CorrelatedObject.Key` and `Correlated.Index` to change to require the `new` when they don't need it, and this API works well for how I want to reuse the comparison for CEL and Schemas.

It is against the contract of `SubPropertyValidator` to call `Validate` on anything but the given property, so I think it is safe to do (same for index). It is `ratchetingValueValidator` is a completely internal implementation that is destroyed at the end of the call to `ValidateUpdate`